src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1691,25 +1697,33 @@ private static int GetArgumentRank(AssociativeNode val)
             return 1 + firstRank;
         }
 
-        private bool WillCallReplicate(List<Type> paramTypes, bool isStaticOrCtor, List<AssociativeNode> args)
+        private bool WillCallReplicate(string fepSig, List<Type> paramTypes, bool isStaticOrCtor, List<AssociativeNode> args)
         {
-            if (isStaticOrCtor)
+            if (willReplicateCache.TryGetValue(fepSig, out bool willReplicate))
             {
-                return !DoesParamArgRankMatch(paramTypes, args);
+                return willReplicate;
```

Comment:
so - if the arguments change won't this value potentially be incorrect?

I feel that either
1. We also need to put the arguments into the signature as the key, OR
2. We need to fallback to replication when making a direct call, maybe using a try catch that gets emitted?

Reply:
Yep, totally forgot about the arguments